### PR TITLE
T346029 ES Module

### PR DESCRIPTION
**Pull request by RickiJay-WMDE**
### Import

`require` statements converted to `import`. I tended to import modules rather than specific functions where possible, which resulted in fewer changes to the main body of most files.

### `__dirname`

`__dirname` doesn't exist in ES Modules. In most cases, we were doing `path.join(__dirname, 'SOME_FILE_NA.ME')` or `__dirname + /SOME_FILE_NA.ME`. The simplest replacement was `new URL('SOME_FILE_NA.ME', import.meta.url)` in almost all cases.

### `*.conf.js`

Dynamic `import` is async, so importing the requested suite's `conf.js` file into `wdio.conf.js` doesn't work. I took the opposite approach - each suite's `conf.js` file imports `wdio.conf.js`, and extends it with its specified `specs`. Then `package.json` actually calls `wdio` using the suite's `conf.js` file, rather than the base `wdio.conf.js`.

### `skipIfExtensionNotPresent`

In my last ticket, I included a `# TODO` wrt the setup for skipping extensions; namely, creating the `installed_extensions` field shouldn't presently work and should be refactored. I included that refactor here; `skipIfExtensionNotPresent()` is now async and fetches the installed extensions, which is admittedly less efficient but should be more future-proof.

### CODE REVIEW

**Review by rti — CHANGES_REQUESTED** [review]
Great work!

File: test/suites/base__fedprops/base__fedprops.conf.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
-'use strict';
+import { deepmerge } from 'deepmerge-ts';
```
[the remark this answers is not in the document]
Comment (RickiJay-WMDE) [reply test/suites/base__fedprops/base__fedprops.conf.js:1]: That does work, but `deepmerge-ts` is already a requirement of `@wdio/runner`, so we're already installing it.

[the remark this answers is not in the document]
Comment (RickiJay-WMDE) [reply test/suites/base__fedprops/base__fedprops.conf.js:1]: Would you still rather we use `lodash`?

**Review by lorenjohnson — APPROVED** [review]
LGTM! 🎉

File: test/suites/base__fedprops/base__fedprops.conf.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rti) [reply test/suites/base__fedprops/base__fedprops.conf.js:1]: Sorry for not responding for a while.

Well, I still see a benefit in keeping our direct dependencies as low as possible. I would have opt'ed for using lodash instead. Anyway. Also happy that this is merged as is.

**Pull request merged**